## amplicon

### tools
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [UPARSE](http://drive5.com/uparse/) | C++/C | v12.0-beta1 | 2024.6 | _Nat Methods_ | 2013.8 | [UPARSE: Highly accurate OTU sequences from microbial amplicon reads](https://doi.org/10.1038/nmeth.2604) |
| [Mothur](https://github.com/mothur/mothur/tree/v1.48.1) | C++ | v1.48.1 | 2024.5 | _Appl Environ Microbiol_ | 2009.12 | [Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities](https://doi.org/10.1128/AEM.01541-09) |
| [dada2](https://github.com/benjjneb/dada2) | R | 1.26 | 2022.11 | _Nat Methods_ | 2016.5 | [DADA2: High resolution sample inference from Illumina amplicon data.](https://doi.org/10.1038/nmeth.3869)
| [QIIME2](https://qiime2.org/) | Java | 2024-05 | 2024.5 | _Nat Biotechnol_ | 2019.7 | [Reproducible, interactive, scalable and extensible microbiome data science using QIIME 2](https://doi.org/10.1038/s41587-019-0209-9) |
| [RDP Classifier](https://sourceforge.net/projects/rdp-classifier/) | Java | 2.14 | 2023.8 | _Microb Ecol_ | 2024.3 | [Updated RDP taxonomy and RDP Classifier for more accurate taxonomic classification](https://doi.org/10.1128/mra.01063-23) |
| [Tax4Fun](https://tax4fun.gobics.de) | R | v0.3.1 | 不再更新 | _Bioinformatics_ | 2015.5 | [Tax4Fun: predicting functional profiles from metagenomic 16S rRNA data](http://dx.doi.org/10.1093/bioinformatics/btv287) |
| Tax4Fun2 | 链接无效 ||| _Environ Microbiome_ | 2020.5 | [Tax4Fun2: prediction of habitat-specific functional profiles and functional redundancy based on 16S rRNA gene sequences](https://doi.org/10.1186/s40793-020-00358-7) |
| [PICRUSt](https://github.com/picrust/picrust) | Python | v1.1.4 | 2019.6 | _Nat Biotechnol_ | 2013.8 | [Predictive functional profiling of microbial communities using 16S rRNA marker gene sequences](https://doi.org/10.1038/nbt.2676) |
| [PICRUSt2](https://github.com/picrust/picrust2) | Python | v2.5.3 | 2024.8 | _Nat Biotechnol_ | 2020.6 | [PICRUSt2 for prediction of metagenome functions.](https://doi.org/10.1038/s41587-020-0548-6) |
| [graftM](https://github.com/geronimp/graftM) | Python | v0.14.0 | 2022.5 | _Nucleic Acids Res_ | 2018.6 | [GraftM: a tool for scalable, phylogenetically informed classification of genes within metagenomes](https://doi.org/10.1093/nar/gky174) |
| [RiboSnake](https://github.com/IKIM-Essen/RiboSnake) | snakemake | v0.10.0 | 2024.8 | _Gigabyte_ | 2024.8 | [RiboSnake – a user-friendly, robust, reproducible, multipurpose and documentation-extensive pipeline for 16S rRNA gene microbiome analysis](https://doi.org/10.46471/gigabyte.132) |
| [ssUMI](https://github.com/ZielsLab/ssUMI) | Shell | NoReleaseTag ||  _PNAS Nexus_ | 2024.10 | [High accuracy meets high throughput for near full-length 16S ribosomal RNA amplicon sequencing on the Nanopore platform](https://doi.org/10.1093/pnasnexus/pgae411) |


### databases
| Database | Type | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------- | ---- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [EUKARYOME](https://eukaryome.org) | 18S/ITS/28S | v1.9.2 | 2024.8 | _Database_ | 2024.6 | [EUKARYOME: the rRNA gene reference database for identification of all eukaryotes](https://doi.org/10.1093/database/baae043) |
| [Silva](https://www.arb-silva.de) | 16S | v138.2 | 2024.7 | _Nucleic Acids Res_ | 2013.1 | [The SILVA ribosomal RNA gene database project: improved data processing and web-based tools](https://doi.org/10.1093/nar/gks1219) |
| [Greengenes2](http://ftp.microbio.me/greengenes_release/) | 16S | v2022.10 | 2022.10 | _Nat Biotechnol_ | 2023.7 | [Greengenes2 unifies microbial data in a single reference tree](https://doi.org/10.1038/s41587-023-01845-1) |
| [MiDAS](https://midasfieldguide.org/guide) | 16S | v5 | 2024.7 | _Nat Commun_ | 2024.6 | [MiDAS 5: Global diversity of bacteria and archaea in anaerobic digesters](https://doi.org/10.1038/s41467-024-49641-y) |