## taxonomy | metagenome profile

### read based
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [MetaPhlAn](https://github.com/biobakery/MetaPhlAn) | Python | v4.1.1 | 2024.5 | _Nat Biotechnol_ | 2023.2 | [Extending and improving metagenomic taxonomic profiling with uncharacterized species using MetaPhlAn 4](https://doi.org/10.1038/s41587-023-01688-w)
| [StrainPhlAn](https://github.com/biobakery/MetaPhlAn) | Python | v4.1.1 | 2024.5 | _Genome Res_ | 2017.2 | [Microbial strain-level population structure and genetic diversity from metagenomes](https://dx.doi.org/10.1101/gr.216242.116)
| [kraken2](https://github.com/DerrickWood/kraken2) | C++ | v2.1.3 | 2023.6 | _Genome Biol_ | 2019.11 | [Improved metagenomic analysis with Kraken 2](https://doi.org/10.1186/s13059-019-1891-0)
| [Metabuli](https://github.com/steineggerlab/Metabuli) | C++ | v1.0.7 | 2024.9 | _Nat Methods_ | 2024.5 | [Metabuli: sensitive and specific metagenomic classification via joint analysis of amino acid and DNA](https://doi.org/10.1038/s41592-024-02273-y)
| [Metabuli-App](https://github.com/steineggerlab/Metabuli-App) | js | v1.0.1 | 2025.1 | bioRxiv | 2025.3 | [Easy and interactive taxonomic profiling with Metabuli App.](https://doi.org/10.1101/2025.03.10.642298) | macOS/Windows/Linux
| [SingleM Microbial Fraction(SMF)](https://github.com/EisenRa/2024_soil_dark_matter_reply) | Python ||| _ISME Commun_ | 2024.9 | [Quantifying microbial DNA in metagenomes improves microbial trait estimation.](https://doi.org/10.1093/ismeco/ycae111)
| [sylph](https://github.com/bluenote-1577/sylph) | Rust | v0.6.1 | 2024.4 | _Nat Biotechnol_ | 2024.10 | [Metagenome profiling and containment estimation through abundance-corrected k-mer sketching with sylph](https://doi.org/10.1038/s41587-024-02412-y)
| [Kaiju](https://github.com/bioinformatics-centre/kaiju) | C | v1.10.1 | 2024.3 | _Nat Commun_ | 2016.4 | [Fast and sensitive taxonomic classification for metagenomics with Kaiju](https://doi.org/10.1038/ncomms11257)
| [mOTUs-db](https://motus-db.org/) | web ||| _Nuclei Acids Res_ | 2024.11 | [The mOTUs online database provides web-accessible genomic context to taxonomic profiling of microbial communities](https://doi.org/10.1093/nar/gkae1004)
| [mOTUs](https://github.com/motu-tool/mOTUs) |||| bioRxiv |2022.4| [Reference genome-independent taxonomic profiling of microbiomes with mOTUs3](https://doi.org/10.1101/2021.04.20.440600)
| [melon](https://github.com/xinehc/melon) | Python | v0.2.4 | 2024.12 ||| [Melon: metagenomic long-read-based taxonomic identification and quantification using marker genes](https://doi.org/10.1186/s13059-024-03363-y)
| [kun-peng](https://github.com/eric9n/Kun-peng) | Rust | v0.7.4 | 2024.9 | bioRxiv | 2024.12 | [Kun-peng: an ultra-memory-efficient, fast,  and accurate pan-domain taxonomic classifier for all](https://doi.org/10.1101/2024.12.19.629356v1)
| [MNBC](https://github.com/ComputationalPathogens/MNBC) | Java | v1.2 | 2025.1 | _Bioinformatics_ | 2024.10 | [MNBC: a multithreaded Minimizer-based Naïve Bayes Classifier for improved metagenomic sequence classification](https://doi.org/10.1093/bioinformatics/btae601)


## contig/sequence based
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [MEGAN6](https://uni-tuebingen.de/fakultaeten/mathematisch-naturwissenschaftliche-fakultaet/fachbereiche/informatik/lehrstuehle/algorithms-in-bioinformatics/software/megan6/) |||| _Biol Direct_ | 2018.4 | [MEGAN-LR: new algorithms allow accurate binning and easy interactive exploration of metagenomic long reads and contigs](https://doi.org/10.1186/s13062-018-0208-7)
| [MMseq2/easy-taxonomy](https://github.com/soedinglab/mmseqs2) | C/C++ | v15-6f452 | 2023.10 | _Bioinformatics_ | 2021.3 | [Fast and sensitive taxonomic assignment to metagenomic contigs](https://doi.org/10.1093/bioinformatics/btab184)
| [CDKAM](https://github.com/SJTU-CGM/CDKAM) | C++/Perl | v1.1 | 2020.11 | _BMC Bioinform_ | 2020.10 | [CDKAM: a taxonomic classification tool using discriminative k-mers and approximate matching strategies](https://doi.org/10.1186/s12859-020-03777-y)
| [MetaMaps](https://github.com/DiltheyLab/MetaMaps) | Perl/C++ ||| _Nat Commun_ | 2019.7 | [Strain-level metagenomic assignment and compositional estimation for long reads with MetaMaps](https://doi.org/10.1038/s41467-019-10934-2)
| [BugSeq](https://app.bugseq.com/academic) | Web Online | 可试用 || _BMC Bioinform_ | 2021.3 | [BugSeq: a highly accurate cloud platform for long-read metagenomic analyses](https://doi.org/10.1186/s12859-021-04089-5)
| [Vamb/Taxometer](https://github.com/RasmussenLab/vamb) | Python | v4.1.3 | 2023.6 | _Nat Commun_ | 2024.9 | [Taxometer: Improving taxonomic classification of metagenomics contigs](https://doi.org/10.1038/s41467-024-52771-y)
| [geNomad](https://github.com/apcamargo/genomad) | Python | v1.8.0 | 2024.4 | _Nat Biotechnol_ | 2023.9 | [Identification of mobile genetic elements with geNomad](https://doi.org/10.1038/s41587-023-01953-y)
| [VITAP](https://github.com/DrKaiyangZheng/VITAP) | Python | v1.7.1 | 2025.2 | _Nat Commun_ | 2025.3 | [VITAP: a high precision tool for DNA and RNA viral classification based on meta-omic data](https://doi.org/10.1038/s41467-025-57500-7)


### MAG based
| Tool/workflows | Language | Lastest version | Release date | Journal | Pub Date | Paper title | Notes |
| -------------- | -------- | --------------- | ------------ | ------- | -------- | ----------- | ----- |
| [GTDB-Tk](https://github.com/ecogenomics/gtdbtk) | Python | v1.7.0 | 2021.10 | _Bioinformatics_ | 2020.3 | [GTDB-Tk: a toolkit to classify genomes with the Genome Taxonomy Database](https://doi.org/10.1093/bioinformatics/btz848)
| [GTDB-Tk 2](https://github.com/Ecogenomics/GTDBTk) | Python | v2.4.0 | 2024.04 | _Bioinformatics_ | 2022.10 | [GTDB-Tk v2: memory friendly classification with the Genome Taxonomy Database.](https://doi.org/10.1093/bioinformatics/btac672)
| [tronko](https://github.com/lpipes/tronko) | C ||| _eLife_ | 2024.9 | [A rapid phylogeny-based method for accurate community profiling of large-scale metabarcoding datasets](https://doi.org/10.7554/eLife.85794)
| [kMetaShot](https://github.com/gdefazio/kMetaShot) | Python || 2024.9 | _Brief Bioinform_ | 2025.1 | [kMetaShot: a fast and reliable taxonomy classifier for metagenome-assembled genomes](https://doi.org/10.1093/bib/bbae680)
